app/models/user.rb

```diff
@@ -121,8 +121,7 @@ def with_team_kind(kind)
     end
 
     def with_all_associations_joined
-      includes(:roles).group("roles.id").
-      includes(roles: :team).group("teams.id")
+      includes(roles: :team).group("roles.id").group("teams.id")
```

Comment:
Rails 5.2.1 didn't like that `roles` was being included twice, and there's no need to include it twice.